docs/configuration/query-frontend.md

```diff
@@ -0,0 +1,240 @@
+## Disclaimer
+
+This aims to be a general purpose example. There are a number of substitutions to make for these to work correctly. These variables take the form of <variable_name>. Override them with specifics to your environment.
+
+## Configuration
+
+Use this shared config file to get the benefits of query parallelisation and caching with the query-frontend component. In addition to this configuration, start the querier and query-frontend components with `-target=querier` and `-target=query-frontend`, respectively.
+
+```yaml
+# Disable the requirement that every request to Cortex has a
+# X-Scope-OrgID header. `fake` will be substituted in instead.
+auth_enabled: false
+
+# We don't want the usual /api/prom prefix.
+http_prefix:
+
+server:
+  http_listen_port: 9091
+
+query_range:
+  # make queries more cache-able by aligning them with their step intervals
+  align_queries_with_step: true
+  max_retries: 5
+  # parallelize queries in 15min intervals
+  split_queries_by_interval: 15m 
+  cache_results: true
+
+  results_cache:
+    max_freshness: 10m
+    cache:
+      # We're going to use the in-process "FIFO" cache, but you can enable
+      # memcached below.
+      enable_fifocache: true
+      fifocache:
+        size: 1024
+        validity: 24h
+
+      # If you want to use a memcached cluster, configure a headless service
+      # in Kubernetes and Cortex will discover the individual instances using
+      # a SRV DNS query.  Cortex will then do client-side hashing to spread
+      # the load evenly.
+      # memcached:
+      #   memcached_client:
+      #     host: memcached.default.svc.cluster.local
+      #     service: memcached
+      #     consistent_hash: true
+
+frontend:
+  # 256 length tenant queues per frontend
+  max_outstanding_per_tenant: 256
+  log_queries_longer_than: 5s
+  compress_responses: true
+  
+frontend_worker:
+  address: query-frontend.<namespace>.svc.cluster.local:9095
```

Comment:
I think advertising http is better. so they don't have to change their querier setup.

Reply:
This is the address for the queriers to connect to and they pull from the frontend over grpc. I don't think the queriers can pull over http.

Reply:
Sorry I wasn't clear. I think we should remove this section. and use the `downstream` property of the frontend config. The frontend will use http to request queriers.

Reply:
The idea is to use the frontend without changing anything else in your stack, just by adding a new service.

This is not the best way to use the frontend, I agree but this could be explained.